Update login guide, add section on VSCode (#76)
mila-iqia/mila-docs: docs/VSCode.rst

.. code-block:: bash

    mila code path/to/code

Note that the above command requires your SSH config to be organized in a certain way,
so you should run ``mila init`` prior to using ``mila code`` for the first time.

If you already have an allocation, or you want to run VSCode on the same node as one
of your running jobs, the ``--node name`` and ``--job jobid`` options can be used.

.. _milatools: https://github.com/mila-iqia/milatools

Activating an environment
-------------------------

Reference_

.. _reference: https://code.visualstudio.com/docs/python/environments

To activate a conda or pip environment, you can open the command palette with Ctrl+Shift+P
and type "Python: Select interpreter". This will prompt you for the path to the Python
executable for your environment.

.. tip::

    If you already have the environment activated in a terminal session, you can run the command
    ``which python`` to get the path for this environment. This path can be pasted into the
    interpreter selection prompt in VSCode to use that same environment.

Troubleshooting
---------------

"Cannot reconnect"
^^^^^^^^^^^^^^^^^^

> When connecting to multiple compute nodes (and/or from multiple computers), some instances may crash with that message because of conflicts in the lock files VSCode installs in ``~/.vscode-server`` (which is shared on all compute nodes through NFS). To fix this issue, you can change this setting in your ``settings.json`` file:

Reviewer: La plupart du contenu wrap à 80 colonnes.

> When connecting to multiple compute nodes (and/or from multiple computers), some instances may crash with that message because of conflicts in the lock files VSCode installs in ``~/.vscode-server`` (which is shared on all compute nodes through NFS). To fix this issue, you can change this setting in your ``settings.json`` file:

Reviewer: `/home/mila` is not NFS.

```suggestion
When connecting to multiple compute nodes (and/or from multiple computers), some instances may crash with that message because of conflicts in the lock files VSCode installs in ``~/.vscode-server`` (which is shared on all compute nodes). To fix this issue, you can change this setting in your ``settings.json`` file:
```

.. code-block:: json

    "remote.SSH.lockfilesInTmp": true

> This will store the necessary lockfiles in ``/tmp`` on the compute nodes (which are not shared through NFS).

Reviewer: Open question: can we use `$SLURM_TMPDIR` instead of `/tmp` for lockfiles?

Reviewer 2: This is a common setup for development environments that house a single vs code install at a shared directory used by multiple users (or multiple vs code instances) at the same time. However, yeah, it could break stuff if you're unlucky I think. Probably requires some more thought behind it but it does solve a problem for now, which is having vs code open from multiple computers at the same time (e.g., my laptop and my desktop).

> This will store the necessary lockfiles in ``/tmp`` on the compute nodes (which are not shared through NFS).

Reviewer: ```suggestion
This will store the necessary lockfiles in ``/tmp`` on the compute nodes (which are local to the node).
```
